### PR TITLE
send back accumulated cost for each matched path

### PR DESCRIPTION
**Pull request by kevinkreiser**
this doesnt yet worry about when there are discontinuities etc but does allow us to get some measure of goodness of a given path. it also doesnt worry about collisions (paths with the same cost) which could be an issue i think

### CODE REVIEW

File: src/meili/map_matcher.cc [hunk 1]
```diff
@@ -372,11 +372,12 @@ MapMatcher::OfflineMatch(const std::vector<Measurement>& measurements, uint32_t
   }
 
   //For k paths
-  std::vector<std::vector<MatchResult>> best_paths;
+  std::map<float, std::vector<MatchResult>> best_paths;
   for(uint32_t i = 0; i < k; ++i) {
     // Get the states for the kth best path its in reverse order
     std::vector<StateId> stateids;
     std::copy(vs_.SearchPath(time), vs_.PathEnd(), std::back_inserter(stateids));
+    auto accumulated_cost = vs_.AccumulatedCost(stateids.front());
```
Comment (kevinkreiser) [inline src/meili/map_matcher.cc]: is this a reasonable way to get at how expensive it was to reach the end state?

Comment (ptpt) [reply src/meili/map_matcher.cc]: Yes, but the first stateid might be invalid; we need to find the first valid stateid.

File: src/meili/map_matcher.cc [hunk 2]
```diff
@@ -416,9 +419,16 @@ MapMatcher::OfflineMatch(const std::vector<Measurement>& measurements, uint32_t
...
       // Copy the interpolated match results into the final set
-      std::copy(interpolated_results.cbegin(), interpolated_results.cend(), std::back_inserter(best_paths.back()));
+      std::copy(interpolated_results.cbegin(), interpolated_results.cend(), std::back_inserter(path));
     }
 
+    // Keep the path and the cost for it, if you cant find a valid state id we just give it a huge cost
+    auto found = std::find_if(stateids.rbegin(), stateids.rend(), [](const StateId& si) {
+      return si.IsValid();
+    });
+    auto accumulated_cost = found != stateids.rend() ? vs_.AccumulatedCost(*found) : MAX_ACCUMULATED_COST;
```
Comment (kevinkreiser) [inline src/meili/map_matcher.cc]: @ptpt i think ive properly gotten the last valid stateids accumulated cost now. what are your thoughts on handling discontinuities. i know i had mentioned adding cost where those occured. do you think that would be best done here or perhaps...

File: src/thor/trace_route_action.cc [hunk 3]
```diff
@@ -145,12 +145,13 @@ std::vector<std::tuple<float, std::vector<thor::MatchResult>, odin::TripPath>> t
...
   if (trace.size() > 0)
     offline_results = matcher->OfflineMatch(trace, best_paths);
 
   // Process each score/match result
-  for (const auto& match_results : offline_results) {
+  for (const auto& scored_match_results : offline_results) {
+    const auto& match_results = scored_match_results.second;
     std::vector<thor::MatchResult> enhanced_match_results;
     odin::TripPath trip_path;
     std::unordered_map<size_t, std::pair<RouteDiscontinuity, RouteDiscontinuity>> route_discontinuities;
```
Comment (kevinkreiser) [inline src/thor/trace_route_action.cc]: @ptpt could we do the penalty down in here where we determine where the discontinuities are. i was thinking that i could based off of the mode and distance between the two points on either side of the discontinuity i could compute a penality? does that make sense at all? what is your preference?

**Pull request closed without merging**